lib/ansible/modules/monitoring/icinga2_feature.py

```diff
@@ -21,49 +25,57 @@
 
 short_description: Manage Icinga2 feature
 description:
-    - Enable or disable an Icinga2 feature
+    - This module can be used to enable or disable an Icinga2 feature.
 version_added: "2.3"
 author: "Loic Blot (@nerzhul)"
 options:
     name:
-        description:
-            - This is the feature name to enable or disable.
-        required: True
+      description:
+      - This is the feature name to enable or disable.
+      required: True
     state:
-        description:
-            - Apply feature state.
-        required: false
-        choices: [ "present", "absent" ]
-        default: present
+      description:
+      - If set to C(present) and feature is disabled, then feature is enabled.
+      - If set to C(present) and feature is already enabled, then nothing is changed.
+      - If set to C(absent) and feature is enabled, then feature is disabled.
+      - If set to C(absent) and feature is already disabled, then nothing is changed.
+      choices: [ "present", "absent" ]
+      default: present
 '''
 
 EXAMPLES = '''
 - name: Enable ido-pgsql feature
   icinga2_feature:
     name: ido-pgsql
     state: present
+
+- name: Disable api feature
+  icinga2_feature:
+    name: api
+    state: absent
 '''
 
 RETURN = '''
 #
 '''
 
 import re
-
 from ansible.module_utils.basic import AnsibleModule
 
 
 class Icinga2FeatureHelper:
     def __init__(self, module):
         self.module = module
         self._icinga2 = module.get_bin_path('icinga2', True)
+        self.feature_name = self.module.params['name']
+        self.state = self.module.params['state']
 
     def _exec(self, args):
         if not self.module.check_mode:
```

Comment:
This condition is the problem. It leads to `icinga2 feature list` never beeing executed in `check_mode`, which results in empty `out`. The condition should be removed alltogether. `self._exec` is only invoked twice anyway: first to run `icinga2 feature list` (should be done regardless of `check_mode` or not) and second at a place in code that is never reached in `check_mode` anyway.